src/Common/tests/System/RealFormatterTestsBase.netcoreapp.cs

```diff
@@ -0,0 +1,326 @@
+        [InlineData(Math.E, "2.718281828459045")]
+        [InlineData(Math.PI, "3.141592653589793")]
+        [InlineData(0.0, "0")]
+        [InlineData(0.84551240822557006, "0.8455124082255701")]
+        [InlineData(1.0, "1")]
+        [InlineData(1844674407370955.25, "1844674407370955.2")]
+        public void TestFormatterDouble_R(double value, string expectedResult) => TestFormatterDouble_Standard(value, "R", expectedResult);
+
+        [Theory]
+        [InlineData(double.Epsilon, "5E-324")]
+        [InlineData(double.MaxValue, "1.7976931348623157E+308")]
+        [InlineData(Math.E, "2.718281828459045")]
+        [InlineData(Math.PI, "3.141592653589793")]
+        [InlineData(0.0, "0")]
+        [InlineData(0.84551240822557006, "0.8455124082255701")]
+        [InlineData(1.0, "1")]
+        [InlineData(1844674407370955.25, "1844674407370955.2")]
+        public void TestFormatterDouble_R4(double value, string expectedResult) => TestFormatterDouble_Standard(value, "R4", expectedResult);
+
+        [Theory]
+        [InlineData(double.Epsilon, "5E-324")]
+        [InlineData(double.MaxValue, "1.7976931348623157E+308")]
+        [InlineData(Math.E, "2.718281828459045")]
+        [InlineData(Math.PI, "3.141592653589793")]
+        [InlineData(0.0, "0")]
+        [InlineData(0.84551240822557006, "0.8455124082255701")]
+        [InlineData(1.0, "1")]
+        [InlineData(1844674407370955.25, "1844674407370955.2")]
+        public void TestFormatterDouble_R20(double value, string expectedResult) => TestFormatterDouble_Standard(value, "R20", expectedResult);
+
+        [Theory]
+        [InlineData(double.Epsilon)]
+        [InlineData(double.MaxValue)]
+        [InlineData(Math.E)]
+        [InlineData(Math.PI)]
+        [InlineData(0.0)]
+        [InlineData(0.84551240822557006)]
+        [InlineData(1.0)]
+        [InlineData(1844674407370955.25)]
+        public void TestFormatterDouble_X(double value) => TestFormatterDouble_Invalid(value, "X");
+
+        [Theory]
+        [InlineData(double.Epsilon)]
+        [InlineData(double.MaxValue)]
+        [InlineData(Math.E)]
+        [InlineData(Math.PI)]
+        [InlineData(0.0)]
+        [InlineData(0.84551240822557006)]
+        [InlineData(1.0)]
+        [InlineData(1844674407370955.25)]
+        public void TestFormatterDouble_X4(double value) => TestFormatterDouble_Invalid(value, "X4");
+
+        [Theory]
+        [InlineData(double.Epsilon)]
+        [InlineData(double.MaxValue)]
+        [InlineData(Math.E)]
+        [InlineData(Math.PI)]
+        [InlineData(0.0)]
+        [InlineData(0.84551240822557006)]
+        [InlineData(1.0)]
+        [InlineData(1844674407370955.25)]
+        public void TestFormatterDouble_X20(double value) => TestFormatterDouble_Invalid(value, "X20");
+
+        private void TestFormatterDouble_Invalid(double value, string format)
+        {
+            Assert.Throws<FormatException>(() => InvariantToStringDouble(value, format));
+        }
```

Comment:
Seems like this could be condensed into something like:
```C#
public static IEnumerable<object[]> InvalidDoubleFormatMemberData =>
    from value in new[] { double.Epsilon, double.MaxValue, … }
    from format in new[] { "X", "X4", "X20" }
    select new object[] { value, format };

[Theory]
[MemberData(nameof(InvalidDoubleFormatMemberData))]
public void InvalidFormat_Throws(double value, string format) => Assert.Throws<FormatException>(() => InvariantToStringDouble(value, format));
```

Reply:
Fixed.